File: front-end/src/app/domain/models/game/action/impl/jail/FreeFromJailAction.ts
import {GameAction} from "../../GameAction";
import {Player} from "../../../Player";

/**
 * TODO: Add JsDoc
 *
 * @author Hamza el Haouti
 */
export class FreeFromJailAction extends GameAction {

  public constructor(
    id: number,
    actor: Player,
    performedAt: Date,
  ) {
    super(FreeFromJailAction.name, id, actor, performedAt);
  }

  public static trueCopy(action: FreeFromJailAction): GameAction | null {
    return new FreeFromJailAction(
      action.id,
      Player.trueCopy(action.actor),
      new Date(action.performedAt),
    );
  }

  public override toString(): string {
    return `${this.actor.user.username} has freed from jail.`;
  }

}

File: back-end/src/main/java/nl/hva/backend/domain/models/game/action/impl/jail/FreeFromJailAction.java
package nl.hva.backend.domain.models.game.action.impl.jail;

import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;
import nl.hva.backend.domain.models.game.Player;
import nl.hva.backend.domain.models.game.action.GameAction;

import javax.persistence.Entity;

/**
 * TODO: Add JavaDoc
 *
 * @author Hamza el Haouti
 */
@Entity
@Getter
@Setter
@NoArgsConstructor
public class FreeFromJailAction extends GameAction {

    public FreeFromJailAction(
            Player actor
    ) {
        super(actor);
    }

}